src/omotes_sdk/omotes_interface.py

```diff
@@ -184,14 +193,68 @@ def connect_to_submitted_job(
         :param auto_disconnect_on_result: Remove/disconnect from all queues pertaining to this job
             once the result is received and handled without exceptions through
             `callback_on_finished`.
+        :param auto_dead_letter_after_ttl: When erroneous situations occur (e.g. client is offline),
+            the job result message (if available) will be dead lettered after the given TTL,
+            and all queues of this job will be removed subsequently. Default to 48 hours if unset.
+            Set to `None` to turn off auto dead letter and clean up, but be aware this may lead to
+            messages and queues to be stored in RabbitMQ indefinitely
+            (which uses up memory & disk space).
+        :param reconnect: When True, first check the job queues status and raise an error if not
+            exist. Default to True.
         """
+        job_results_queue_name = OmotesQueueNames.job_results_queue_name(job.id)
+        job_progress_queue_name = OmotesQueueNames.job_progress_queue_name(job.id)
+        job_status_queue_name = OmotesQueueNames.job_status_queue_name(job.id)
+
+        if reconnect:
+            logger.info("Reconnect to the submitted job %s is set to True. "
+                        + "Checking job queues status...", job.id)
+            if not self.broker_if.queue_exists(job_results_queue_name):
+                raise RuntimeError(
+                    f"The {job_results_queue_name} queue does not exist or is removed. "
+                    "Abort reconnecting to the queue."
+                )
+            if (callback_on_progress_update
+                    and not self.broker_if.queue_exists(job_progress_queue_name)):
+                raise RuntimeError(
+                    f"The {job_progress_queue_name} queue does not exist or is removed. "
+                    "Abort reconnecting to the queue."
+                )
+            if (callback_on_status_update
+                    and not self.broker_if.queue_exists(job_status_queue_name)):
+                raise RuntimeError(
+                    f"The {job_status_queue_name} queue does not exist or is removed. "
+                    "Abort reconnecting to the queue."
+                )
+
         if auto_disconnect_on_result:
             logger.info("Connecting to update for job %s with auto disconnect on result", job.id)
             auto_disconnect_handler = self._autodelete_progres_status_queues_on_result
         else:
             logger.info("Connecting to update for job %s and expect manual disconnect", job.id)
             auto_disconnect_handler = None
 
+        # TODO: handle reconnection after the message is dead lettered but queue still exists.
+
+        if auto_dead_letter_after_ttl is not None:
+            message_ttl = auto_dead_letter_after_ttl
+            queue_ttl = auto_dead_letter_after_ttl * 2
```

Comment:
I removed `JOB_QUEUES_REMOVAL_BUFFER` (previously set too small) and set `queue_ttl` to be two times of `auto_dead_letter_after_ttl` 

**Why**: If the worker is running for a long time (assuming almost but slightly less than `auto_dead_letter_after_ttl` time), the job result message might be returned to the queue where the queue is soon to expire. The message will be dropped silently due to the expired queue.

Reply:
If I understand the rational correctly, then no SDK will have connected to this queue for a period of time. Lets call this time `queue_offline_time`. If we set `queue_ttl` == `auto_dead_letter_after_ttl + 30 seconds`, then the queue will be deleted after the SDK has been offline in that period of time. We are assuming that a period of 48 hours (`auto_dead_letter_after_ttl`) is sufficient to say that an error scenario has occurred and no SDK will retrieve any other messages from this queue. Wouldn't it then hold that `auto_dead_letter_after_ttl` is sufficient time that the queue may be expired and any results are dropped silently?

I am making the assumption here that connecting to a queue is sufficient to not trigger the TTL. Lets say the queue TTL is set to 2 hours. Do you know if RabbitMQ also triggers the queue TTL and expires it if the queue has been empty for 2 hours but a consumer was connected in all that time?

Reply:
Let's say SDK is never reconnected, and we want the job result message is _at least (which is not the case here)_ dead lettered and logged by the orchestrator and queues to be cleaned up in the end. Then the issue raised here is that the job result message could actually be dropped silently due to the expired queue.

1. SDK submits a job and declares queues (assume `auto_dead_letter_after_ttl = 1hr`, and `queue_ttl = auto_dead_letter_after_ttl + 30 seconds`)
2. Worker starts processing the job (assume taking 50 mins to complete)
3. Meanwhile, SDK is offline 1 mins after the job submission due to errors. Queue TTL starts counting down. The queue will be expired in 1 hr + 30 sec 
4. The worker finishes the job (took 50 mins) and the result is returned to the queue. The job result will be dead lettered in 1 hr. However, the job queue is actually expired first before the result can be dead lettered.

_"I am making the assumption here that connecting to a queue is sufficient to not trigger the TTL."_ -> You are right on this. Queue TTL is activated and starts counting down when there is no consumer (SDK offline). When the queue has a consumer (SDK reconnects), the Queue TTL is basically inactive.

Reply:
Ahh gotcha! Does publishing the message to the queue not refresh the Queue TTL? Or are only consumer-side subscriptions counted towards refreshing a queue TTL?

Reply:
I had a misconception that republishing a message to the queue resets Queue TTL, but it turned out not the case...